tests/cloud_functions/testing_utils.py

```
@@ -0,0 +1,224 @@
+
+    def update(self, data):
+        """Updates the document's data with the provided data.
+
+        Args:
+            data (dict): The data to update in the document.
+
+        Raises:
+            Exception: If the document does not exist.
+        """
+        if not self.exists:
+            raise Exception("Document does not exist")
```

Ruff (0.8.2)

Create your own exception

(TRY002)